reconnecting_ftp/__init__.py

```diff
@@ -112,7 +113,10 @@ def connect(self) -> None:
         if self.connection is None:
             conn_refused = None  # type: Optional[ConnectionRefusedError]
             try:
-                self.connection = self.FTP(timeout=self.timeout, encoding=self.encoding)
+                if sys.version_info >= (3, 9):
```

Comment:
(Unfortunately GitHub does not let me comment on unchanged lines. This comment refers to line 77.)

I only now realized that the user might be confused if the `encoding` parameter is simply ignored on older versions of Python.

What do you think about this additional change:
```if sys.version_info >= (3, 9):
    def __init__(self,
             hostname: str,
             port: int,
             user: str,
             password: str,
             max_reconnects: int = 10,
             timeout: int = 10,
             FTP=ftplib.FTP,
             encoding: str = 'utf-8') -> None:
        # pylint: disable=too-many-arguments
        self.access = Access()
        self.access.hostname = hostname
        self.access.port = port
        self.access.user = user
        self.access.password = password
        self.connection = None  # type: Optional[ftplib.FTP]
        self.last_pwd = None  # type: Optional[str]
        self.encoding = encoding
        self.max_reconnects = max_reconnects
        self.timeout = timeout
        self.FTP = FTP  # pylint: disable=invalid-name
else:
    def __init__(self,
             hostname: str,
             port: int,
             user: str,
             password: str,
             max_reconnects: int = 10,
             timeout: int = 10,
             FTP=ftplib.FTP,
             encoding: str = 'utf-8') -> None:
        # pylint: disable=too-many-arguments
        self.access = Access()
        self.access.hostname = hostname
        self.access.port = port
        self.access.user = user
        self.access.password = password
        self.connection = None  # type: Optional[ftplib.FTP]
        self.last_pwd = None  # type: Optional[str]
        
        # No ``encoding `` is supported 
        # for this version of Python.
        
        self.max_reconnects = max_reconnects
        self.timeout = timeout
        self.FTP = FTP  # pylint: disable=invalid-name
```

Reply:
I didn't quite understand your intention in your code snippet, but I understood your suggestion that we not silently ignore a value provided for the encoding argument.

How about this latest update?  It only raises an error IF a value is supplied for the encoding argument and the version < 3.9.  If no argument is provided, the self.encoding attribute is never set.  If the self.encoding attribute is not set, it is not passed to the call to self.FTP( ).  This allows ftplib.FTP() to be responsible for setting the default value of encoding, and reconnecting-ftp doesn't have to provide a default at all.